### PR TITLE
Keep track of neccessary scroll pages to skip useless last request

### LINKED ISSUE #1234: Unnecessary request to ES when iterating over ResultSets using Scroll
There is one unnecessary call to ES API when using Elastica\Scroll. The last call (in next method) always sets `$this->_currentResultSet` with empty Elastica\ResultSet. It could be omitted.
The solution is quite simple. In rewind method number of total pages could be count
`$this->totalPages = ceil($scroll->current()->getTotalHits() / $scroll->current()->count());`

Then in next method request should be sent only if there is next page `$this->currentPage <= $this->totalPages;`
In next method `$this->currentPage` should be incremented.

### PR DESCRIPTION
**Pull request by Tobion**
Fixes #1234

### CODE REVIEW

File: lib/Elastica/Scroll.php [hunk 1]
```diff
@@ -32,12 +32,14 @@ class Scroll implements \Iterator
 
     /**
      * 0: scroll<br>
-     * 1: scroll id<br>
-     * 2: search type.
+     * 1: scroll id
      *
      * @var array
      */
-    protected $_options = [null, null, null];
```
Comment (ruflin) [inline lib/Elastica/Scroll.php]: search type was never used?

Comment (Tobion) [inline lib/Elastica/Scroll.php]: It got removed before already with es upgrade

**Comment by Tobion** [comment]
Added changelog

**Comment by ruflin** [comment]
@Tobion Thanks, merged.

**Pull request merged**